.gitlab/check.yml

```diff
@@ -0,0 +1,31 @@
+.check_skip_rust:
+  before_script:
+    - |-
+      if echo "$CI_COMMIT_TITLE" | grep -q "ci-skip-rust"; then
+        echo "Skipping rust tests due to commit title"
+        exit 0
+      fi
+
+.check_skip_integration_tests:
+  before_script:
+    - |-
+      if echo "$CI_COMMIT_TITLE" | grep -q "ci-skip-integration-tests"; then
+        echo "Skipping integration tests due to commit title"
+        exit 0
+      fi
+
+cargo-clippy:
+  extends: .check_skip_rust
+  image: paritytech/ci-unified:bullseye-1.77.0
+  stage: check
+  cache:
+    key: ${CI_COMMIT_REF_SLUG}-clippy
```

Comment:
Why this cache key is different than what's used for build and test steps?

Reply:
Also, since we use a different combination of flags, one with default features and one with `--all-features`, is it possible to have two different cache entries? Otherwise they would interfere with each other.

Reply:
That would mean splitting the job into two separate jobs which I believe would be overkill.

Reply:
That would mean splitting the job into two separate jobs which I believe would be overkill.

Reply:
That would mean splitting the job into two separate jobs which I believe would be overkill.